client_inventory.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+type InventoryReporter interface {
+	Submit(api ApiRequester, server string, data interface{}) error
+}
+
+type InventoryClient struct {
+}
+
+func NewInventoryClient() InventoryReporter {
+	return &InventoryClient{}
+}
+
+// Report status information to the backend
+func (i *InventoryClient) Submit(api ApiRequester, url string, data interface{}) error {
```

Comment:
Look at for example `Upload` method in `client_log.go`. This is exact same code with the difference of how we are making request. Maybe we should unify this somehow instead of copy-paste. 
What I am thinking is:

```
type MakeRequest func(url string, data interface{}) (*http.Request, error)

type Requester interface {
	Request(api ApiRequester, server string, data interface{}, makeFunc MakeRequest)
}
```
We can also skip `data interface{}` above and assume `data` will be a part of receiver on which we will call `MakeRequest()` function.